### PR TITLE
Feature/xpansion  split mps

### PR DESCRIPTION
**Pull request by guilpier-code**
Antares for Xpansion : split MPS into fixed and variable part.

When asking Xpansion mode, the first week of the simulation dumps the fixed part of the first optimization problem on disk, and every week dumps the its own variable part of the optimization problem.

This PR regards more specially the split the division ot MPS data into its fixed part and variable part.

This PR is related to issue #121

### CODE REVIEW

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 1]
```diff
@@ -548,6 +560,221 @@ void OPT_EcrireResultatFonctionObjectiveAuFormatTXT(void * Prob, uint numSpace,
...
+			}
+
+			il++;
+		}
+	}
+
+	free( Cder );
+
+	auto& study = *Data::Study::Current::Get();
+	Flot = study.createFileIntoOutputWithExtension("fixed-part", "mps", numSpace);
```
Comment (flomnes) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp]: `createFileIntoOutputWithExtension` assumes that prefix (the first argument, here "fixed-part") is either "problem" or "criterion", see mps.cpp:65 for example.

Therefore file numbering for other prefixes such as "fixed-part" is not handled properly. Consequently, `createFileIntoOutputWithExtension` needs to be updated to handle prefix="fixed-part".

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 2]
```diff
@@ -548,6 +560,221 @@ void OPT_EcrireResultatFonctionObjectiveAuFormatTXT(void * Prob, uint numSpace,
...
+		{
+			fprintf(Flot, " L  R%07d\n", Cnt);
+		}
+		else if (Pb->Sens[Cnt] == '>')
+		{
+			fprintf(Flot, " G  R%07d\n", Cnt);
+		}
+		else
+		{
+			fprintf(Flot, "Writing fixed part of MPS data : le sens de la contrainte %c ne fait pas partie des sens reconnus\n", Pb->Sens[Cnt]);
```
Comment (flomnes) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp]: Don't mix French and English

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 3]
```diff
@@ -548,6 +560,221 @@ void OPT_EcrireResultatFonctionObjectiveAuFormatTXT(void * Prob, uint numSpace,
...
+
+	if (Nombre == NULL)
+	{
+		logs.fatal() << "Not enough memory";
+		AntaresSolverEmergencyShutdown();
+	}
+
+	// gp : to be changed
+	auto& study = *Data::Study::Current::Get();
+	Flot = study.createFileIntoOutputWithExtension("variable-part", "mps", numSpace);
```
Comment (flomnes) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp]: Same remark, prefix "variable-part" is not handled properly

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/mps.cpp [hunk 4]
```diff
@@ -40,73 +40,70 @@ namespace Antares
...
-		logs.info();
-                return fd;
-        }
-
-
+FILE* Study::createFileIntoOutputWithExtension(const YString& prefix,
+                                               const YString& extension,
+                                               uint numSpace) const
+{
+    static std::map<YString, int> count;
```
Comment (guilpier-code) [inline src/libs/antares/study/mps.cpp:45]: ok, i approve the fix made by @flomnes

**Review by flomnes — APPROVED** [review]

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 5]
```diff
@@ -446,25 +677,51 @@ void OPT_EcrireJeuDeDonneesLineaireAuFormatMPS(void* Prob, uint numSpace, char T
...
+            ExistenceDUneSolution = OUI_ANTARES;
+
+        NombreDeVariables = ProblemePourPne->NombreDeVariables;
+        TypeDeBorneDeLaVariable = ProblemePourPne->TypeDeBorneDeLaVariable;
+        Xmax = ProblemePourPne->Xmax;
+        Xmin = ProblemePourPne->Xmin;
+        X = ProblemePourPne->X;
+        CoutLineaire = ProblemePourPne->CoutLineaire;
+        NombreDeContraintes = ProblemePourPne->NombreDeContraintes;
+        SecondMembre = ProblemePourPne->SecondMembre;
```
Comment (flomnes) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp]: PNE has been removed from the develop branch, this distinction is no longer necessary.

**Pull request closed without merging**